src/core/Ros.js

```diff
@@ -61,15 +61,15 @@ export default class Ros extends EventEmitter {
         socketAdapter(this)
       );
     } else if (this.transportLibrary === 'websocket') {
-      // Detect if in browser vs in NodeJS
-      if (typeof window !== 'undefined') {
+      // browsers, Deno, and Bun support WebSockets natively
+      if (typeof WebSocket == "function") {
```

Comment:
Please fix the failing checks on this line to adhere to the project's style guidelines